src/renderer/components/common/Scanning/ScanMultiframeQr.tsx

```diff
@@ -62,9 +68,20 @@ const ScanMultiframeQr = ({
       return (async () => {
         const { payload, unsigned } = await createPayload(transactions[index], api);
 
+        const signPayload =
```

Comment:
I guess `createSubstrateSignPayload` works better and already contains needed condition to determine the right payload creator function